rocketpy/motors/tank.py

```diff
@@ -1292,7 +1292,20 @@ def fluid_volume(self):
         Function
             Volume of the fluid as a function of time.
         """
-        return self.liquid_volume + self.gas_volume
+        fluid_volume = self.liquid_volume + self.gas_volume
+
+        # Check if within bounds
+        diff = fluid_volume - self.geometry.total_volume
+
+        if (diff > 1e-6).any():
+            raise ValueError(
+                "The `fluid_volume`, defined as the sum of `gas_volume` and "
+                + "`liquid_volume`, is not equal to the total volume of the tank."
+                + "\n\t\tThe difference is more than 1e-6 m^3 at "
+                + f"{diff.x_array[np.argmin(diff.y_array)]} s."
```

Comment:
Just checking: we need to guarantee, for now, that `fluid_volume` does not exceed `self.geometry.total_volume`, but we don't really care yet if `fluid_volume` is much less then `self.geometry.total_volume`?

If this is the case, shouldn't we us `np.argmax` instead of `np.argmin`?

Reply:
I do think you are right regarding `np.argmax`, but, just to make sure we are on the same page, this would need to be changed in all error messages of this class, right?

Reply:
Actually, I correct myself: the computation should be `np.argmin` since we are loking for the smallest of the differences between the fluid volume and the tank volume to determine exactly where it overfilled.

Reply:
@phmbressan with `np.argmin`, don't you have the risk of getting a negative value, in which the tank is overfilled?

Reply:
The check here is made following these assumptions:

1. The `diff` pointed out that the `Tank` is, at some time, overfilled;
2. There is a value of time that the fluid volume surpassed the total volume of the Tank;
3. In general, this specific point in time is not part of the time samples so the minimum value of the diff is used to select it. I do think it is possible that this value may be a bit before the overfill crossing or a bit after, so there is a certain margin of error.

Perhaps I should mention that the time is approximate or, likely a better solution, use `Function.find_input`. I favored the implemented way due the fact this is a hotfix and the `np.argmin` was already being used in the other exceptions of this class.

Let me know if I made it clearer now or if there is any mistake in my reasoning.